### PR TITLE
Implement Lucene's deprecated RegexQuery

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by mfussenegger** [comment]
Did you do some benchmarks to compare old vs new?

**Comment by matriv** [comment]
@mfussenegger No I forgot :-( I will do.

**Comment by matriv** [comment]
`+----------------------------------------------------------------------------------------------------------+----------------------+----------------------+-----------------------+
| run		 | statement                                                                                   | runtime_stats['max'] | runtime_stats['min'] | runtime_stats['mean'] |
+----------------------------------------------------------------------------------------------------------+----------------------+----------------------+-----------------------+
| deprecated | select * from userVisits where "UserAgent" ~ '(?i)\bMozilla/5.0\s\(Windows\b.*' limit 10000 |            413.56073 |            252.47214 |          320.81100028 |
| customImpl | select * from userVisits where "UserAgent" ~ '(?i)\bMozilla/5.0\s\(Windows\b.*' limit 10000 |            378.1013  |            211.54988 |          257.44308    |
| deprecated | select * from userVisits where "UserAgent" ~ '.*Windows\sNT\s6.0.*' limit 10000             |            602.938   |            214.11084 |          263.30870054 |
| customImpl | select * from userVisits where "UserAgent" ~ '.*Windows\sNT\s6.0.*' limit 10000             |            582.4014  |            210.12956 |          259.81630242 |
| deprecated | select * from userVisits where "UserAgent" ~ '.*\bWindows\sNT\b.*' limit 10000              |            435.21085 |            246.03607 |          320.40040666 |
| customImpl | select * from userVisits where "UserAgent" ~ '.*\bWindows\sNT\b.*' limit 10000              |            380.98834 |            210.2292  |          254.632152   |
+----------------------------------------------------------------------------------------------------------+----------------------+----------------------+-----------------------+`

**Comment by matriv** [comment]
Benchmarks results:
https://gist.github.com/matriv/7295084d066cfe42f8dbafa9ecd42fbc

File: sql/src/main/java/io/crate/lucene/match/CrateRegexCapabilities.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+package io.crate.lucene.match;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.CharsRefBuilder;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An implementation tying Java's built-in java.util.regex to {@link CrateRegexQuery}.
```
Comment (seut) [inline sql/src/main/java/io/crate/lucene/match/CrateRegexCapabilities.java]: can we document why we did this here or somewhere else pls?

File: sql/src/main/java/io/crate/lucene/match/CrateRegexQuery.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.util.AttributeSource;
+
+import java.io.IOException;
+
+
+/**
+ * Implements the regular expression term search query.
+ */
```
Comment (seut) [inline sql/src/main/java/io/crate/lucene/match/CrateRegexQuery.java]: pls document why we did this

**Pull request merged**